var/spack/repos/builtin/packages/nvhpc/package.py

```diff
@@ -386,6 +387,24 @@ def setup_dependent_package(self, module, dependent_spec):
             self.spec.mpif77 = join_path(mpi_prefix.bin, "mpif77")
             self.spec.mpifc = join_path(mpi_prefix.bin, "mpif90")
 
+    @property
+    def headers(self):
+        prefix = Prefix(
+            join_path(self.prefix, "Linux_%s" % self.spec.target.family, self.version, "compilers")
+        )
+        headers = []
+        if "+blas" in self.spec:
+            headers.append("cblas.h")
+        if "+lapack" in self.spec:
+            headers.append("lapacke.h")
+
+        if "+ilp64" in self.spec:
+            prefix = prefix.include.ilp64
+        else:
+            prefix = prefix.include.lp64
+
+        return find_headers(headers, prefix)
+
     @property
     def libs(self):
```

Comment:
This still needs to fix the libraries to support forcing ilp64 and also needs the ability to detect the system default for the headers.

Once that is done, there needs to be a fix to set up the build/runtime environments correctly.